tests/test_predictions.py

```diff
@@ -20,6 +20,18 @@
     "https://raw.githubusercontent.com/vincentarelbundock/Rdatasets/master/csv/ggplot2/diamonds.csv"
 )
 
+def test_newdata_balanced():
+    df = pl.read_csv(
+        "https://vincentarelbundock.github.io/Rdatasets/csv/palmerpenguins/penguins.csv",
+        null_values="NA",
+    ).drop_nulls()
+    mod = smf.ols(
+        "body_mass_g ~ flipper_length_mm * species * bill_length_mm + island",
+        df.to_pandas(),
+    ).fit()
+    pre_py = predictions(mod, newdata="balanced")
+    pre_r = pl.read_csv("tests/r/test_predictions_newdata_balanced_01.csv")
+    compare_r_to_py(pre_r, pre_py)
```

Comment:
These tests are very "expensive", both in developer time and in runtime. If we can, we should just test things in raw Python. Here, we don't need to test the actual numeric values, since we know that `predictions()` is fine (from other tests). All we need to check is if we have 9 combinations of categorical predictors. So you can have a simpler assertion without doing the whole `compare_r_to_py()`